src/main/java/seedu/todo/commons/util/FilterUtil.java

```diff
@@ -0,0 +1,346 @@
+package seedu.todo.commons.util;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+
+import seedu.todo.models.Event;
+import seedu.todo.models.Task;
+
+/**
+ * Helper function to help in filtering results
+ * 
+ * @@author A0139922Y
+ *
+ */
+
+public class FilterUtil {
+    
+    /*==================== Filtering Methods for Tasks ======================*/
+    
+    /**
+     * Filter the task list based on matching task name list
+     * @param tasks 
+     *             Provided list for filtering
+     * @param namelist
+     *             Search and filter based on the name list
+     */
+    public static List<Task> filterTaskByNames(List<Task> tasks, HashSet<String> nameList) {
+        //if search list size is 0 , user not searching based on name
+        List<Task> filteredTasks = new ArrayList<Task>();
+        if (nameList.size() == 0) {
+            return filteredTasks;
+        }
+        
+        for (int i = 0; i < tasks.size(); i ++) {
+            Task task = tasks.get(i);
+            Iterator<String> nameListIterator = nameList.iterator();
+            while (nameListIterator.hasNext()) {
+                String currentMatchingName = nameListIterator.next().toLowerCase();
+                if (task.getName().toLowerCase().startsWith(currentMatchingName)) {
+                    filteredTasks.add(task);
+                    break;
+                }
+                
+                boolean matchFound = false;
+                String [] taskNameBySpace = StringUtil.splitStringBySpace(task.getName());
+                for (int j = 0; j < taskNameBySpace.length; j ++) {
+                    if (taskNameBySpace[j].toLowerCase().startsWith(currentMatchingName)) {
+                        filteredTasks.add(task);
+                        matchFound = true;
+                        break;
+                    }
+                }
+                
+                if (matchFound) {
+                    break;
+                }
+            }
+            nameListIterator = nameList.iterator();
+        }
+        return filteredTasks;
+    }
```

Comment:
SLAP can be improved in this method. Try to reduce nesting. For example, you can extract a method that searches for one name and then call that method in a loop from a higher level method.